File: flutter/lib/utils/log_helper.dart
import 'package:flutter/foundation.dart';

class LogHelper {
  static const String _tag = "RecebimentoApp";

  static void d(String message) {
    if (kDebugMode) debugPrint("$_tag D: $message");
  }

  static void e(String message, [Object? error, StackTrace? stackTrace]) {
    if (kDebugMode) {
      debugPrint("$_tag E: $message");
      if (error != null) debugPrint(error.toString());
    }
  }

  static void i(String message) {
    if (kDebugMode) debugPrint("$_tag I: $message");
  }

  static void w(String message) {
    if (kDebugMode) debugPrint("$_tag W: $message");
  }
}

File: app/src/main/java/io/recebimento/utils/LogHelper.kt
package io.recebimento.utils

import android.util.Log

object LogHelper {
    private const val TAG = "RecebimentoApp"
    
    fun d(message: String) {
        Log.d(TAG, message)
    }
    
    fun e(message: String, throwable: Throwable? = null) {
        if (throwable != null) {
            Log.e(TAG, message, throwable)
        } else {
            Log.e(TAG, message)
        }
    }
    
    fun i(message: String) {
        Log.i(TAG, message)
    }
    
    fun w(message: String) {
        Log.w(TAG, message)
    }
}